netgear/netgear.go

```
@@ -0,0 +1,165 @@
+package netgear
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const timeout = 10 * time.Second
+
+type Netgear struct {
+	sessionToken string
+	client       *http.Client
+
+	baseUrl  url.URL
+	username string
+	password string
+}
+
+func NewNetgear(hostName, username, password string) *Netgear {
+	u, _ := url.Parse(strings.TrimSpace(hostName))
+	if u == nil {
+		u = &url.URL{}
+	}
```

Same here! If the user provides a URL that can't even be parsed why ignore it silently? This function should also return a `(*Netgear, err)` and you should return the error returned from `url.Parse` instead of ignoring it.